pystruct/inference/inference_methods.py

```diff
@@ -4,23 +4,32 @@
 from .maxprod import inference_max_product
 from .common import _validate_params
 
-
 def get_installed(method_filter=None):
     if method_filter is None:
-        method_filter = ["max-product", 'ad3', 'qpbo', 'ogm', 'lp']
+        method_filter = ["max-product", 'ad3', 'ad3+', 'qpbo', 'ogm', 'lp']
 
     installed = []
     unary = np.zeros((1, 1))
     pw = np.zeros((1, 1))
     edges = np.empty((0, 2), dtype=np.int)
     for method in method_filter:
         try:
-            inference_dispatch(unary, pw, edges, inference_method=method)
+            if method != 'ad3+':
+                inference_dispatch(unary, pw, edges, inference_method=method)
+            else:
+                inference_dispatch(unary, np.zeros((0,1,1)), np.zeros((0,2), dtype=np.int), inference_method=method)
```

Comment:
can't you handle this in the ad3+ wrapper instead of here?

Reply:
ad3+ input has a different input structure, so we need to change from the default ones chosen in lines 13 - 15.
Apart from this ad3+ is handled the same as the other methods, so it makes sense to have this code here.